postgresql/provider.go

```diff
@@ -60,6 +60,12 @@ func Provider() terraform.ResourceProvider {
 				Optional:   true,
 				Deprecated: "Rename PostgreSQL provider `ssl_mode` attribute to `sslmode`",
 			},
+			"connection_string": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("PGCONNSTRING", nil),
+				Description: "The libPQ style connection string for the database to connect to",
```

Comment:
Specifying the `connection_string` parameter should conflict with the `host` parameter.  In fact, this should conflict with most of the existing parameters.